### PR TITLE
Add HTTP Header Filter documentation.

### PR DESCRIPTION
**Pull request by t4lz**
Should only be merged once https://github.com/metalbear-co/mirrord/pull/843 is merged and released.

### CODE REVIEW

File: content/docs/reference/traffic/index.md [hunk 1]
```diff
@@ -165,6 +165,33 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
+remote target.
+For each `Header-Name`, `Header-Value` pair, your regular expression will be matched against `Header-Name:
+Header-Value`. For example, the filter `MyHeader` would match requests with `MyHeader` in any of their header names
+or header values. The filter `^X-MyFilter:` would match only requests that have a header with the header name
+`X-MyFilter` (or `x-myfilter` or with any other capitalization).
+The regular expression is evaluated with the [fancy_regex](https://docs.rs/fancy-regex/0.10.0/fancy_regex/index.html)
+rust crate.
+
+###### Specifying a Filter
+The HTTP header filter can be specified in the mirrord configuration by setting the incoming mode to
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:182]: We don't want users to use the env var, so lets remove any mentions of it in the docs

File: content/docs/reference/traffic/index.md [hunk 2]
```diff
@@ -165,6 +165,33 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
 </tr>
 </table>
 
+##### Filtering Incoming Traffic by HTTP Headers
+
+Currently only supported in `steal` mode: mirrord lets you specify a regular expression to filter HTTP requests with.
+When specified, all the headers of each HTTP request that arrives at the "impersonated" target are checked against
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:171]:
```suggestion
When specified, all the headers of each HTTP request that arrives at the remote target are checked against
```

You use remote later anyway, so at least this way it's clearer you're talking about the same thing

File: content/docs/reference/traffic/index.md [hunk 3]
```diff
@@ -165,6 +165,33 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
+`MIRRORD_HTTP_HEADER_FILTER` environment variable.
+A filter cannot be specified via the command line.
+
+###### Setting Custom HTTP Ports
+The configuration also allows specifying custom HTTP ports under `feature.network.incoming.http_header_filter.ports`,
+or via the `MIRRORD_HTTP_HEADER_FILTER_PORTS` environment variable. By default, ports 80 and 8080 are used as HTTP ports
+if a filter is specified, which means that the mirrord agent checks each new connection on those ports for HTTP, and if
+the connection has valid HTTP messages, they are filtered with the header filter.
+
+Currently only supporting HTTP version 1.
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:193]:
```suggestion
Currently only HTTP version 1 is supported.
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: content/docs/reference/traffic/index.md [hunk 4]
```diff
@@ -165,6 +165,32 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
+remote target.
+For each `Header-Name`, `Header-Value` pair, your regular expression will be matched against `Header-Name:
+Header-Value`. For example, the filter `MyHeader` would match requests with `MyHeader` in any of their header names
+or header values. The filter `^X-MyFilter:` would match only requests that have a header with the header name
+`X-MyFilter` (or `x-myfilter` or with any other capitalization).
+The regular expression is evaluated with the [fancy_regex](https://docs.rs/fancy-regex/0.10.0/fancy_regex/index.html)
+rust crate.
+
+###### Specifying a Filter
+The HTTP header filter can be specified in the mirrord configuration by setting the incoming mode to
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:182]:
```suggestion
The HTTP header filter can be specified in the mirrord configuration file by setting the incoming mode to
```

File: content/docs/reference/traffic/index.md [hunk 5]
```diff
@@ -165,6 +165,32 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
+`steal` and specifying a filter in `feature.network.incoming.http_header_filter.filter`.
+A filter cannot be specified via an argument in the command line interface.
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:184]: I don't think we should mention this

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: content/docs/reference/traffic/index.md (hunk 5 above)
Comment (t4lz) [reply content/docs/reference/traffic/index.md:184]:
```suggestion
```

**Review by eyalb181 — APPROVED** [review]

**Pull request merged**